applications/ShapeOptimizationApplication/python_scripts/design_logger_vtk.py

```diff
@@ -6,71 +6,96 @@
-        if OutputMode == "WriteDesignSurface":
-            self.WriteDesignSurface = True
-        elif OutputMode == "WriteOptimizationModelPart":
-            if self.OptimizationModelPart.NumberOfElements() == 0:
-                raise NameError("Output of optimization model part in VKT-format requires definition of elements. No elements are given in current mdpa! You may change the design output mode.")
-            self.WriteOptimizationModelPart = True
-        else:
-            raise NameError("The following design output mode is not defined within a VKT output (name may be misspelled): " + OutputMode)
+        self.model = model_part_controller.GetModel()
+        self.optimization_model_part = model_part_controller.GetOptimizationModelPart()
+        self.design_surface = model_part_controller.GetDesignSurface()
+
+        self.vtk_io = self.__CreateVTKIO()
 
     # --------------------------------------------------------------------------
     def __CreateVTKIO( self ):
-        ResultsDirectory = self.OutputSettings["output_directory"].GetString()
-        DesignHistoryFilename = self.OutputSettings["design_history_filename"].GetString()
-        DesignHistoryFilenameWithPath =  ResultsDirectory+"/"+DesignHistoryFilename
 
-        NodalResults = self.OutputSettings["nodal_results"]
+        vtk_parameters = self.output_settings["output_format"]["Parameters"]
 
-        if self.WriteDesignSurface:
-            self.VtkIO = VTKFileIO( self.DesignSurface, DesignHistoryFilenameWithPath, "WriteConditionsOnly", NodalResults )
-        elif self.WriteOptimizationModelPart:
-            self.VtkIO = VTKFileIO( self.OptimizationModelPart, DesignHistoryFilenameWithPath, "WriteElementsOnly", NodalResults )
+        nodal_results = self.output_settings["nodal_results"]
+        vtk_parameters.AddValue("nodal_solution_step_data_variables", nodal_results)
+
+        output_mode = self.output_settings["design_output_mode"].GetString()
+        if output_mode == "WriteDesignSurface":
+            vtk_parameters["model_part_name"].SetString(self.design_surface.Name)
+        elif output_mode == "WriteOptimizationModelPart":
+            vtk_parameters["model_part_name"].SetString(self.optimization_model_part.Name)
+        else:
+            raise NameError("The following design output mode is not defined within a VTK output (name may be misspelled): " + output_mode)
+
+        vtk_parameters["folder_name"].SetString(self.output_settings["output_directory"].GetString())
+
+        return vtk_output_process.Factory(self.output_settings["output_format"], self.model)
 
     # --------------------------------------------------------------------------
     def InitializeLogging( self ):
-        self.VtkIO.InitializeLogging()
+        self.vtk_io.ExecuteInitialize()
+        self.vtk_io.ExecuteBeforeSolutionLoop()
 
     # --------------------------------------------------------------------------
     def LogCurrentDesign( self, optimizationIteration ):
```

Comment:
I think its a good idea to stick to the style guide regarding the naming of variabels. E.g. optimizationIteration -> optimization_iteration (function argument) or below OriginalTime -> original_time (local variable).